### PR TITLE
Issue 278: Fix segment number

### LINKED ISSUE #278: Segment number does not start from 0
**Problem description**
If a stream recreated right after deletion, the segment number may start from a non-zero number due to the lazy deletion of segment in tier2. So if user provides a segment 0 to the Byte Writer, what it means is actually segment x.

**Problem location**
Byte Writer

**Suggestions for an improvement**

### PR DESCRIPTION
**Pull request by Tristan1900**
**Change log description**  
Byte stream API now takes a fixed size stream instead of a paticular segment as the input. This stream is a special stream that could only contains one segment.

**Purpose of the change**  
FIx #278

**What the code does**  
Use stream instead of segment as byte stream API input. Ensure that the stream contains only one segment.

**How to verify it**  
Tests should pass.

### CODE REVIEW

**Review by tkaitchuck — CHANGES_REQUESTED** [review]
I don't think we want to adjust the number given to us by users. We should instead just take the stream name and work it out from the controller.

**Review by shrids — APPROVED** [review]
LGTM apart from the two errors in the tests.

File: integration_test/src/lib.rs [hunk 1]
```diff
@@ -94,12 +94,12 @@ mod test {
             .unwrap();
 
         // metric::metric_init(PROMETHEUS_SCRAPE_PORT.parse::<SocketAddr>().unwrap());
-        info!("Running integration test");
-        let config = PravegaStandaloneServiceConfig::new(false, true, true);
-        run_tests(config);
-
-        let config = PravegaStandaloneServiceConfig::new(true, false, false);
-        run_tests(config);
+        // info!("Running integration test");
```
Comment (tkaitchuck) [inline integration_test/src/lib.rs:97]: What's going on here?

**Review by tkaitchuck — COMMENTED** [review]
This looks good. But it's changing a public API so you'll need to bump the version number in the toml.

File: integration_test/src/lib.rs (hunk 1 above)
Comment (Tristan1900) [reply integration_test/src/lib.rs]: fixed!

**Review by tkaitchuck — APPROVED** [review]

**Pull request merged**